sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -46,19 +47,35 @@ case class UnscaledValue(child: Expression) extends UnaryExpression {
  */
 case class MakeDecimal(child: Expression, precision: Int, scale: Int) extends UnaryExpression {
 
+  private val nullOnOverflow = SQLConf.get.decimalOperationsNullOnOverflow
+  private lazy val doEval = if (nullOnOverflow) {
```

Comment:
Is there a performance advantage to caching the result of the `nullOnOverflow` check instead of putting an `if-else` in `def nullSafeEval` itself?

My intuition (possibly wrong) is that a method call indirecting through `doEval` is going to be more expensive than a simple boolean check inside the body of `nullSafeEval` (or at least comparable).

If we don't have a strong motivation for this approach, I have a preference for the simpler:

```scala
protected override def nullSafeEval(input: Any): Any = {
  if (nullOnOverflow) {
    new Decimal().setOrNull(input.asInstanceOf[Long], precision, scale)
  } else {
    new Decimal().set(input.asInstanceOf[Long], precision, scale)
  }
}
```

because it more closely matches the `doGenCode`.

Reply:
Usually the focus on perf is a bit less stressed in the evaluated code since we should never execute it, while it is critical for the codegen case. I don't have a strong preference. I can go with your suggestion, thanks.